### PR TITLE
280 Refactoring of extended options in Appconfigs

### PR DESCRIPTION
**Pull request by Maddi02**
Replaced the old OnlyOffice configuration with a new structured approach using ExtendedOptionKeys and ExtendedOptionField. Updated associated files and components to align with the new config structure for better maintainability and clarity.

### CODE REVIEW

File: apps/frontend/src/pages/Settings/AppConfig/AppConfigPage.tsx [hunk 1]
```diff
@@ -75,12 +73,16 @@ const AppConfigPage: React.FC = () => {
       label: item.label,
     }));
 
-    const newExtendedOptions = currentConfig.extendedOptions?.map((item) => ({
-      name: item.name,
-      value: item.value,
-      description: item.description,
-      type: item.type,
-    }));
+    const newExtendedOptions = Array.isArray(currentConfig.extendedOptions)
```
Comment (TomlDev) [inline apps/frontend/src/pages/Settings/AppConfig/AppConfigPage.tsx:76]: Why can `currentConfig.extendedOptions` be something different than an array or undefined?

File: apps/frontend/src/pages/Settings/AppConfig/AppConfigPage.tsx [hunk 2]
```diff
@@ -75,12 +73,16 @@ const AppConfigPage: React.FC = () => {
...
+      ? currentConfig.extendedOptions
+          .filter((item) => item && item.name)
+          .map((item) => ({
+            name: item.name,
+            value: item.value || '',
+            description: item.description || '',
+            type: item.type || '',
+          }))
+      : [];
```
Comment (TomlDev) [inline apps/frontend/src/pages/Settings/AppConfig/AppConfigPage.tsx:85]: Why is this whole `newExtendedOptions` necessary?

File: apps/frontend/src/pages/Settings/AppConfig/AppConfigPage.tsx [hunk 3]
```diff
@@ -125,15 +127,7 @@ const AppConfigPage: React.FC = () => {
...
-    const extendedOptions =
-      settingLocation === 'filesharing'
-        ? (appExtendedOptions.ONLY_OFFICE.map((e) => ({
-            name: e.name,
-            description: e.description,
-            type: e.type,
-            value: getValues(`${settingLocation}.${e.name}`) as string,
-          })) as AppConfigExtendedOption[])
-        : [];
+    const extendedOptions = form.getValues(`${settingLocation}.extendedOptions`) as AppConfigOptions[];
```
Comment (TomlDev) [inline apps/frontend/src/pages/Settings/AppConfig/AppConfigPage.tsx:130]: Type casting is wrong, should be `AppConfigExtendedOption`

File: apps/frontend/src/pages/FileSharing/hooks/useOnlyOffice.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ const useOnlyOffice = ({ filePath, fileName, url, type, mode }: UseOnlyOfficePro
   const fileExtension = getFileExtension(fileName);
   const editorType = useMemo(() => findDocumentsEditorType(fileExtension), [fileExtension]);
   const { appConfigs } = useAppConfigsStore();
-  const documentServerURL = getExtendedOptionValue(appConfigs, appExtendedOptions, AppExtendedOptions.ONLY_OFFICE_URL);
+  const documentServerURL = getExtendedOptionValue(appConfigs, ExtendedOptionKeys.ONLY_OFFICE_URL);
```
Comment (TomlDev) [inline apps/frontend/src/pages/FileSharing/hooks/useOnlyOffice.ts:30]: grafik

File: apps/frontend/src/pages/Settings/AppConfig/components/ExtendedOptionsForm.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            autoComplete="new-password"
+          />
+        );
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <form
```
Comment (TomlDev) [inline apps/frontend/src/pages/Settings/AppConfig/components/ExtendedOptionsForm.tsx:66]: grafik

File: apps/frontend/src/pages/Settings/AppConfig/AppConfigPage.tsx [hunk 6]
```diff
@@ -116,42 +98,34 @@ const AppConfigPage: React.FC = () => {
...
-    const extendedOptions =
-      settingLocation === 'filesharing'
-        ? (appExtendedOptions.ONLY_OFFICE.map((e) => ({
-            name: e.name,
-            description: e.description,
-            type: e.type,
-            value: getValues(`${settingLocation}.${e.name}`) as string,
-          })) as AppConfigExtendedOption[])
-        : [];
+    const extendedOptions = form.getValues(`${settingLocation}.extendedOptions`) as ExtendedOptionKeysDto;
```
Comment (michidold) [inline apps/frontend/src/pages/Settings/AppConfig/AppConfigPage.tsx:110]: I think setting it to a empty object is safer than typecasting:
`const extendedOptions = form.getValues(`${settingLocation}.extendedOptions`) || {};`

File: apps/frontend/src/pages/Settings/AppConfig/appConfigOptions.ts [hunk 7]
```diff
@@ -64,7 +67,24 @@ export const APP_CONFIG_OPTIONS: AppConfigOption[] = [
...
+          value: '',
+        },
+        {
+          name: ExtendedOptionKeys.ONLY_OFFICE_JWT_SECRET,
+          title: 'appExtendedOptions.onlyOfficeJwtSecretTitle',
+          description: 'appExtendedOptions.onlyOfficeJwtSecretDescription',
+          type: ExtendedOptionField.password,
+          value: '',
+        },
+      ],
```
Comment (michidold) [inline apps/frontend/src/pages/Settings/AppConfig/appConfigOptions.ts:86]: Is this intended to be the place, to define all extendedOptions? I would define these in another file. Its a constant and could be placed in libs

File: apps/frontend/src/pages/Settings/AppConfig/appConfigSchema.ts
Comment (michidold) [inline apps/frontend/src/pages/Settings/AppConfig/appConfigSchema.ts:1]: We should consider adding some type of validation. Can you add a issue?

File: apps/frontend/src/pages/Settings/AppConfig/components/AppConfigFormField.tsx
Comment (michidold) [inline apps/frontend/src/pages/Settings/AppConfig/components/AppConfigFormField.tsx:1]: We have still a custom FormField component, can't we use that in here? At least we must use the `autoComplete='new-password` option on the input field to prohibit the browser of inserting the safed creds of the user

File: libs/src/appconfig/constants/extendedOptionKeys.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+enum ExtendedOptionKeys {
+  ONLY_OFFICE_URL = 'ONLY_OFFICE_URL',
+  ONLY_OFFICE_JWT_SECRET = 'ONLY_OFFICE_JWT_SECRET',
+}
+
+export default ExtendedOptionKeys;
```
Comment (michidold) [inline libs/src/appconfig/constants/extendedOptionKeys.ts:6]: I would prefere a const object:
```typescript
const ExtendedOptionKeys = {
  ONLY_OFFICE_URL: 'ONLY_OFFICE_URL',
  ONLY_OFFICE_JWT_SECRET: 'ONLY_OFFICE_JWT_SECRET',
} as const;

export default ExtendedOptionKeys;

```

**Review by michidold — CHANGES_REQUESTED** [review]
Looks good over all. Some minor suggestions

File: apps/frontend/src/pages/Settings/AppConfig/appConfigSchema.ts
Comment (TomlDev) [reply apps/frontend/src/pages/Settings/AppConfig/appConfigSchema.ts:1]: Yes https://github.com/edulution-io/edulution-ui/issues/302

File: apps/frontend/src/pages/Settings/AppConfig/components/AppConfigFormField.tsx
Comment (TomlDev) [reply apps/frontend/src/pages/Settings/AppConfig/components/AppConfigFormField.tsx:1]: Yes we have that component but it's cause many issues if you try to use it.
I already spent some hours trying different things but it's a little messed up, so we decided to create a new one that will work.

**Review by michidold — APPROVED** [review]
Looks good

**Pull request merged**